modules/indexing/src/test/java/org/apache/ignite/util/GridCommandHandlerInterruptCommandTest.java

```diff
@@ -0,0 +1,258 @@
+    }
+
+    /**
+     * Creates predifened query entity.
+     *
+     * @return Query entity.
+     */
+    private QueryEntity createQueryEntity() {
+        QueryEntity qryEntity = new QueryEntity();
+        qryEntity.setKeyType(Integer.class.getTypeName());
+        qryEntity.setValueType(UserValue.class.getName());
+        qryEntity.setTableName("USER_TEST_TABLE");
+
+        LinkedHashMap<String, String> fields = new LinkedHashMap<>();
+        fields.put("x", "java.lang.Integer");
+        fields.put("y", "java.lang.Integer");
+        fields.put("z", "java.lang.Integer");
+        qryEntity.setFields(fields);
+
+        LinkedHashMap<String, Boolean> idxFields = new LinkedHashMap<>();
+
+        QueryIndex idx2 = new QueryIndex();
+        idx2.setName("IDX_2");
+        idx2.setIndexType(QueryIndexType.SORTED);
+        idxFields = new LinkedHashMap<>();
+        idxFields.put("x", false);
+        idx2.setFields(idxFields);
+
+        QueryIndex idx3 = new QueryIndex();
+        idx3.setName("IDX_3");
+        idx3.setIndexType(QueryIndexType.SORTED);
+        idxFields = new LinkedHashMap<>();
+        idxFields.put("y", false);
+        idx3.setFields(idxFields);
+
+        QueryIndex idx4 = new QueryIndex();
+        idx4.setName("IDX_4");
+        idx4.setIndexType(QueryIndexType.SORTED);
+        idxFields = new LinkedHashMap<>();
+        idxFields.put("z", false);
+        idx4.setFields(idxFields);
+
+        qryEntity.setIndexes(Arrays.asList(idx2, idx3, idx4));
+        return qryEntity;
+    }
+
+    /**
+     * User value.
+     */
+    private static class UserValue {
+        /** X. */
+        private int x;
+
+        /** Y. */
+        private int y;
+
+        /** Z. */
+        private int z;
+
+        /**
+         * @param x X.
+         * @param y Y.
+         * @param z Z.
+         */
+        public UserValue(int x, int y, int z) {
+            this.x = x;
+            this.y = y;
+            this.z = z;
+        }
+
+        /**
+         * @param seed Seed.
+         */
+        public UserValue(long seed) {
+            x = (int)(seed % 6991);
+            y = (int)(seed % 18679);
+            z = (int)(seed % 31721);
+        }
+
+        /** {@inheritDoc} */
+        @Override public String toString() {
+            return "UserValue{" +
+                "x=" + x +
+                ", y=" + y +
+                ", z=" + z +
+                '}';
+        }
+    }
+
+    /**
+     * Cehcks that validate_indexes command will cancel after it interrupted.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testValidateIndexesCommand() throws Exception {
+        lnsrLog = new ListeningTestLogger(false, log);
+
+        IgniteEx ignite = startGrid(0);
+
+        ignite.cluster().active(true);
+
+        IgniteCache cache = ignite(0).cache(DEFAULT_CACHE_NAME);
+
+        for (int i = 0; i < 500_000; i++)
+            cache.put(i, new UserValue(i));
```

Comment:
Please use datastreamer for initial load.

Reply:
Done.